src/researchhub_case/related_models/external_author_claim_case_model.py

```diff
@@ -0,0 +1,39 @@
+from django.db import models
+
+from researchhub_case.constants.case_constants import (
+    APPROVED,
+    AUTHOR_CLAIM_CASE_STATUS,
+    INITIATED,
+)
+from researchhub_case.related_models.researchhub_case_abstract_model import (
+    AbstractResearchhubCase,
+)
+from researchhub_case.tasks import celery_add_author_citations
+
+
+class ExternalAuthorClaimCase(AbstractResearchhubCase):
+    h_index = models.IntegerField(default=0, null=True)
+    publication_count = models.IntegerField(default=0, null=True)
+    semantic_scholar_id = models.CharField(max_length=16, null=True)
+    google_scholar_id = models.CharField(max_length=16, null=True)
+    status = models.CharField(
+        choices=AUTHOR_CLAIM_CASE_STATUS,
+        default=INITIATED,
+        max_length=32,
+        null=False,
+    )
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["google_scholar_id", "requestor"],
+                name="unique_google_scholar_id_claim",
+            )
+        ]
+
+    def approve_google_scholar(self):
+        celery_add_author_citations.apply_async(
```

Comment:
A lot of tasks are being called from the model, but specifically in the view. So when an editor approves a claim, it will hit the endpoint and inside the view, it will call this model function.